File: Anul 2/Medii de Programare si Proiectare/Lab/mpp-proiect-java-Mothyque/backend/Server/src/main/java/org/example/service/ClientService.java
package org.example.service;

import org.example.domain.Client;
import org.example.domain.Ticket;
import org.example.repository.ClientRepository;

public class ClientService extends Service<Integer, Client>
{
    private final ClientRepository clientRepository;

    public ClientService(ClientRepository clientRepository)
    {
        super(clientRepository);
        this.clientRepository = clientRepository;
    }

    public Client findByNameAndAddress(String name, String address)
    {
        return clientRepository.findClientByNameAndAddress(name, address);
    }
}

File: Anul 2/Medii de Programare si Proiectare/Lab/mpp-proiect-csharp-Mothyque/ProiectMPP/ProiectMPP.Server/Service/ClientService.cs
using System.Collections.Generic;
using ProiectMPP.Common.Domain;
using ProiectMPP.Server.Repository;

namespace ProiectMPP.Server.Service
{
    public class ClientService : Service<int, Client>
    {
        private readonly IClientRepository _clientRepository;

        public ClientService(IClientRepository clientRepository)
            : base(clientRepository)
        {
            this._clientRepository = clientRepository;
        }

        public Client FindByNameAndAddress(string name, string address)
        {
            return _clientRepository.FindClientByNameAndAddress(name, address);
        }
    }
}